sdl_android/src/main/java/com/smartdevicelink/api/VideoStreamingManager.java

```diff
@@ -0,0 +1,571 @@
+	 * @param parameters Video streaming parameters including: codec which will be used for streaming (currently, only
+	 *                    VideoStreamingCodec.H264 is accepted), height and width of the video in pixels.
+	 * @param encrypted Specify true if packets on this service have to be encrypted
+	 */
+	private void startVideoStreaming(VideoStreamingParameters parameters, boolean encrypted){
+		IVideoStreamListener streamListener = startVideoStream(parameters, encrypted);
+		if(streamListener == null){
+			Log.e(TAG, "Error starting video service");
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			return;
+		}
+		VideoStreamingCapability capability = (VideoStreamingCapability) internalInterface.getCapability(SystemCapabilityType.VIDEO_STREAMING);
+		if(capability != null && capability.getIsHapticSpatialDataSupported()){
+			hapticManager = new HapticInterfaceManager(internalInterface);
+		}
+		try {
+			encoder.init(this.context.get(),streamListener,parameters);
+			//We are all set so we can start streaming at at this point
+			encoder.start();
+			//Encoder should be up and running
+			createRemoteDisplay(encoder.getVirtualDisplay());
+			stateMachine.transitionToState(StreamingStateMachine.STARTED);
+		} catch (Exception e) {
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			e.printStackTrace();
+		}
+		Log.d(TAG, parameters.toString());
+	}
+
+	/**
+	 * Stops streaming service and remote display encoder if applicable.
+	 */
+	public void stopStreaming(){
+		if(remoteDisplay!=null){
+			remoteDisplay.stop();
+			remoteDisplay = null;
+		}
+		if(encoder!=null){
+			encoder.shutDown();
+		}
+		if(internalInterface!=null){
+			internalInterface.stopVideoService();
+		}
+		stateMachine.transitionToState(StreamingStateMachine.STOPPED);
+	}
+
+	/**
+	 * Stops video streaming service and removes service listener.
+	 */
+	public void dispose(){
+		stopStreaming();
+
+		// Remove listeners
+		internalInterface.removeServiceListener(SessionType.NAV, serviceListener);
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_TOUCH_EVENT, touchListener);
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_HMI_STATUS, hmiListener);
+
+		stateMachine.transitionToState(StreamingStateMachine.NONE);
+	}
+
+	// PUBLIC METHODS FOR CHECKING STATE
+
+	/**
+	 * Check if a video service is currently active
+	 * @return boolean (true = active, false = inactive)
+	 */
+	public boolean isVideoConnected(){
+		return (stateMachine.getState() == StreamingStateMachine.READY) ||
+				(stateMachine.getState() == StreamingStateMachine.STARTED) ||
+				(stateMachine.getState() == StreamingStateMachine.STOPPED);
+	}
+
+	/**
+	 * Check if video streaming has been paused due to app moving to background
+	 * @return boolean (true = not paused, false = paused)
+	 */
+	public boolean isVideoStreamingPaused(){
```

Comment:
See previous comment on naming; `isPaused`, `isStreamingPaused`, etc

Reply:
Will change to `isPaused`